packages/react-chat/src/components/Chat/ChatBody.tsx

```diff
@@ -44,30 +44,58 @@ export function ChatBody({
   activeChannelId,
 }: ChatBodyProps) {
   const narrow = useNarrow();
-  const [showChannelsList, setShowChannels] = useState(false);
+  const [showChannelsList, setShowChannelsList] = useState(false);
   const [showMembersList, setShowMembersList] = useState(false);
+  const className = useMemo(() => (narrow ? "narrow" : ""), [narrow]);
+
+  const switchChannelList = () => {
+    if (!showChannelsList && showMembersList) {
+      setShowChannelsList(true);
+      setShowMembersList(false);
+    }
+    if (!showChannelsList && !showMembersList) {
+      setShowChannelsList(true);
+    }
+  };
+
+  const switchMemberList = () => {
+    if (!showChannelsList && !showMembersList) {
+      setShowMembersList(true);
+    }
+    if (!showChannelsList && showMembersList) {
+      setShowMembersList(false);
+    }
+    if (showChannelsList && !showMembersList) {
+      setShowChannelsList(false);
+      setShowMembersList(true);
+    }
+  };
```

Comment:
```suggestion
  const switchMemberList = useCallback(() => {
      setShowChannelsList(false);
      setShowMembersList(!showMembersList);
  },[showMembersList]);
```

I think you always want to hide channels list when you switch members list